lib/documents.js

```diff
@@ -2263,12 +2264,143 @@ function writeAllDocumentsImpl(options) {
   let operation = new Operation(
       'read forestInfo', this.client, requestOptions, 'empty', 'single'
   );
-  operation.stream = inputStream;
-  let result = requester.startRequest(operation);
-  operation.response = result;
+
+  let jobState = {
+    docInstance : this,
+    stream: inputStream,
+    requesterCount: 0,
+    docsWrittenSuccessfully:0,
+    docsFailedToBeWritten: 0,
+    jobOptions: jobOptions
+  };
+
+  operation.jobState = jobState;
+
+  requester.startRequest(operation).result(mlutil.callbackOn(operation, onWriteAllInit))
+      .catch((err) => operation.stream.emit('error', err));
+
   return inputStream;
 }
 
+function finishWriter(operation) {
+  operation.jobState.requesterCount--;
+
+  if(operation.jobState.requesterCount<1) {
+
+    if(operation.jobState.jobOptions && operation.jobState.jobOptions.onCompletion){
+
+      let summary = ('Number of documents written successfully = '+
+          ((operation.jobState.docsWrittenSuccessfully>0)?operation.jobState.docsWrittenSuccessfully:0)+'\n'+
+          'Number of documents failed to be written = '+
+          ((operation.jobState.docsFailedToBeWritten>0)?operation.jobState.docsFailedToBeWritten:0)
+      );
```

Comment:
Instead of assembling a string, it would be more useful to provide the application with an object that has a property for each value, similar to the following sketch:

```
    const summary = {
        docsWritten: ...,
        docsFailed: ...
        };
     if (jobState.error) {
        summary.error = ... ;
    }
```

Reply:
If I recall correctly, a later commit switched from the string to an object.

Unfortunately, I haven'f found a way to get GitHub to bundle up the commits as a single diff so the review can be more consistent.  Have you?